src/Renci.SshNet/SftpClient.cs

```diff
@@ -2003,14 +2003,19 @@ private IEnumerable<SftpFile> InternalListDirectory(string path, Action<int> lis
             if (!basePath.EndsWith("/"))
                 basePath = string.Format("{0}/", fullPath);
 
-            var result = new List<SftpFile>();
+            var result = new List<ISftpFile>();
 
             var files = _sftpSession.RequestReadDir(handle);
 
             while (files != null)
             {
-                result.AddRange(from f in files
-                                select new SftpFile(_sftpSession, string.Format(CultureInfo.InvariantCulture, "{0}{1}", basePath, f.Key), f.Value));
+                result.AddRange(
+                    from f in files
+                    select (ISftpFile) new SftpFile(
+                        _sftpSession,
+                        string.Format(CultureInfo.InvariantCulture, "{0}{1}", basePath, f.Key),
+                        f.Value)
+                );
```

Comment:
Can you remove the usage of Linq here, and at the same time get rid of the cast.

Reply:
done